### ISSUE TITLE
no-async-without-await fails with overloads

### ISSUE THREAD

**Issue by danielrentz** [palantir/tslint#4776]
### Bug Report

- __TSLint version__: 5.18
- __TypeScript version__: 3.5.2
- __Running TSLint via__: grunt-tslint

#### TypeScript code being linted

Cannot be reproduced in the playground. As long as the rule is present in the config, the linter does not update while typing code.

```ts
export async function f(i: number): Promise<void>;
export async function f(s1: string, s2: string): Promise<void>;
export async function f(..._args: unknown[]): Promise<void> {
    return Promise.resolve();
}

export class A {
    public async f(i: number): Promise<void>;
    public async f(s1: string, s2: string): Promise<void>;
    public async f(..._args: unknown[]): Promise<void> {
        return Promise.resolve();
    }
}
```

with `tslint.json` configuration:

```json
{
    "extends": ["tslint:recommended"],
    "rules": { "no-async-without-await": true }
}
```

#### Actual behavior

Rule "no-async-without-await" fails at overload signatures.

#### Expected behavior

Rule "no-async-without-await" lets pass them, and only checks the implementation.

**Comment by danielrentz** [comment]
ah, found #4773
